pkg/proxy/ipvs/README.md

```diff
@@ -10,30 +10,220 @@ Linux kernel.
 IPVS runs on a host and acts as a load balancer in front of a cluster of real servers. IPVS can direct requests for TCP
 and UDP-based services to the real servers, and make services of real servers appear as virtual services on a single IP address.
 
+## IPVS vs. IPTABLES
+IPVS mode was introduced in Kubernetes v1.8 and goes beta in v1.9. IPTABLES mode was added in v1.1 and become the default operating mode since v1.2. Both IPVS and IPTABLES are based on `netfilter`.
+Differences between IPVS mode and IPTABLES mode are mainly as follows:
+
+1. IPVS uses hash table as the underlying data structure and works in kernel spaces.
+
+2. IPVS provides better scalability and performance for large clusters. 
+
+3. IPVS supports more sophisticated load balancing algorithms than iptables (least load, least connections, locality, weighted, etc.).  
+
+4. IPVS supports server health checking and connection retries, etc.
+ 
+### When ipvs falls back to iptables
+IPVS proxier will employ iptables in doing packet filtering, SNAT and supporting NodePort type service. Specifically, ipvs proxier will fall back on iptables in the following 4 scenarios.
+
+**1. kube-proxy start with --masquerade-all=true**
+
+If kube-proxy start with `--masquerade-all=true`, ipvs proxier will masquerade all traffic accessing service Cluster IP, which behaves same as what iptables proxier does. Suppose there is a serivice with Cluster IP `10.244.5.1` and port `8080`, then the iptables installed by ipvs proxier should be like what is shown below.
```

Comment:
Nit: "behaves the same as what iptables proxier does" -> "behaves the sames as the iptables proxier"